# M. Model Inference
___

In [13]:
# Import Libraries
import pandas as pd
import pickle

In [14]:
# Loading model
with open('best_model_pred.pkl', 'rb') as file_1:
  best_model = pickle.load(file_1)

## 1. Load the Inference Data

In [15]:
# Data Inference
data = {
        'Order_ID': [4200, 4321, 4384, 5432, 5580],
        'Order_Date': ['01/08/2021','02/28/2021','04/15/2021','09/16/2021','10/18/2021'],
        'Customer_Name': ['Daniel Hamilton', 'Macaronois Croissania', 'Arnold Schuzzenburg', 'Maria Castello', 'Carbonnara'],
        'City': ['Eindhoven', 'Paris', 'Berlin', 'Madrid', 'Rome'],
        'Country': ['Netherlands', 'France', 'Germany', 'Spain', 'Italy'],
        'State': ['North Brabant', 'Ile-de-France', 'Berlin', 'Madrid', 'Lazio'],
        'Region': ['Central', 'Central', 'Central', 'South', 'South'],
        'Segment': ['Home Office', 'Corporate', 'Home Office', 'Consumer', 'Corporate'],
        'Category': ['Office Supplies', 'Office Supplies', 'Furniture', 'Technology', 'Office Supplies'],
        'Ship_Mode': ['Economy Plus', 'Immediate', 'Immediate', 'Economy', 'Economy Plus'],
        'Sub_Category': ['Art', 'Paper', 'Chairs', 'Accessories', 'Binders'],
        'Product_Name': ['BIC Pencil Sharpener, Fluorescent', 'Premium A4 Paper', 'Ergo Chair', 'USB-C Hub', '3-Ring Binder'],
        'Quantity': [2, 100, 1, 5, 20],
        'Cost': [28, 50, 150, 25, 10],
        'Sales': [30, 80, 250, 45, 15]
    }

data_inference = pd.DataFrame(data)
data_inference



Order_ID  Order_Date          Customer_Name       City      Country  \
0      4200  01/08/2021        Daniel Hamilton  Eindhoven  Netherlands   
1      4321  02/28/2021  Macaronois Croissania      Paris       France   
2      4384  04/15/2021    Arnold Schuzzenburg     Berlin      Germany   
3      5432  09/16/2021         Maria Castello     Madrid        Spain   
4      5580  10/18/2021             Carbonnara       Rome        Italy   

           State   Region      Segment         Category     Ship_Mode  \
0  North Brabant  Central  Home Office  Office Supplies  Economy Plus   
1  Ile-de-France  Central    Corporate  Office Supplies     Immediate   
2         Berlin  Central  Home Office        Furniture     Immediate   
3         Madrid    South     Consumer       Technology       Economy   
4          Lazio    South    Corporate  Office Supplies  Economy Plus   

  Sub_Category                       Product_Name  Quantity  Cost  Sales  
0          Art  BIC Pencil Sharpener, Fluorescent         2    28     30  
1        Paper                   Premium A4 Paper       100    50     80  
2       Chairs                         Ergo Chair         1   150    250  
3  Accessories                          USB-C Hub         5    25     45  
4      Binders                      3-Ring Binder        20    10     15

## 2. Prediction Result

In [16]:
# Make the prediction
prediction = best_model.predict(data_inference)

predicted_profit = {
        'Profit': prediction,
        'Order_ID': [4200, 4321, 4384, 5432, 5580],
        'Order_Date': ['01/08/2021','02/28/2021','04/15/2021','09/16/2021','10/18/2021'],
        'Customer_Name': ['Daniel Lambert', 'Macaronois Croissania', 'Arnold Schuzzenburg', 'Maria Castello', 'Carbonnara'],
        'City': ['Eindhoven', 'Paris', 'Berlin', 'Madrid', 'Rome'],
        'Country': ['Netherlands', 'France', 'Germany', 'Spain', 'Italy'],
        'State': ['North Brabant', 'Ile-de-France', 'Berlin', 'Madrid', 'Lazio'],
        'Region': ['Central', 'Central', 'Central', 'South', 'South'],
        'Segment': ['Home Office', 'Corporate', 'Home Office', 'Consumer', 'Corporate'],
        'Category': ['Office Supplies', 'Office Supplies', 'Furniture', 'Technology', 'Office Supplies'],
        'Ship_Mode': ['Economy Plus', 'Immediate', 'Immediate', 'Economy', 'Economy Plus'],
        'Sub_Category': ['Art', 'Paper', 'Chairs', 'Accessories', 'Binders'],
        'Product_Name': ['BIC Pencil Sharpener, Fluorescent', 'Premium A4 Paper', 'Ergo Chair', 'USB-C Hub', '3-Ring Binder'],
        'Quantity': [2, 100, 1, 5, 20],
        'Cost': [28, 50, 150, 25, 10],
        'Sales': [30, 80, 250, 45, 15]
    }

predicted_profit = pd.DataFrame(predicted_profit)
predicted_profit

Profit  Order_ID  Order_Date          Customer_Name       City  \
0    3.26      4200  01/08/2021         Daniel Lambert  Eindhoven   
1   29.88      4321  02/28/2021  Macaronois Croissania      Paris   
2  107.22      4384  04/15/2021    Arnold Schuzzenburg     Berlin   
3   18.76      5432  09/16/2021         Maria Castello     Madrid   
4    4.47      5580  10/18/2021             Carbonnara       Rome   

       Country          State   Region      Segment         Category  \
0  Netherlands  North Brabant  Central  Home Office  Office Supplies   
1       France  Ile-de-France  Central    Corporate  Office Supplies   
2      Germany         Berlin  Central  Home Office        Furniture   
3        Spain         Madrid    South     Consumer       Technology   
4        Italy          Lazio    South    Corporate  Office Supplies   

      Ship_Mode Sub_Category                       Product_Name  Quantity  \
0  Economy Plus          Art  BIC Pencil Sharpener, Fluorescent         2   
1     Immediate        Paper                   Premium A4 Paper       100   
2     Immediate       Chairs                         Ergo Chair         1   
3       Economy  Accessories                          USB-C Hub         5   
4  Economy Plus      Binders                      3-Ring Binder        20   

   Cost  Sales  
0    28     30  
1    50     80  
2   150    250  
3    25     45  
4    10     15

Conclusion from Inference Results:

- The model's predictions on this new set of future orders (transaction) provide valuable insight on the profit. The results confirm that Profit is not driven by high sales, quantity neither cost alone, but by a complex interaction between customer segment, product category, geographical market, and time frame pattern included.

Key takeaway from predictions:

- Furniture's Profitability Confirmed: The Furniture sale in Berlin (Central Region) is predicted to be very profitable 107.22 (in dollar). This validates the pattern that Furniture is a high-performing category, especially when sold to Home Office clients in Central region specifically.

- The Low-Margin Challenge with Office Supplies: The model's predictions highlight a crucial business challenge. Both Office Supplies orders using economy plus for shipping mode, despite one having very high quantity and sales (Binders sales in South Region), are predicted to generate lower profits (3.26 dollar and 4.48 dollar).

Strategic Recommendations based on the predictions:

- Focus on High-Margin Sales: the predictions clearly show that the company's highest returns will come from focusing sales and marketing efforts on selling Furniture for Home Office or Corporate clients in Central regions using specifically Immediate shipping mode.

- Re-evaluate Office Supplies Strategy: while necessary for a full product catalog, the Office Supplies category needs a strategic business review. The company should investigate ways to improve margins, either through cost-saving measures, adjusting pricing for corporate clients, or bundling these items with higher-profit products.